create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/test/utils/ScopusGenerator.java

```diff
@@ -154,14 +161,87 @@ private static BibrecordTp randomBibRecord() {
     }
 
     private static HeadTp randomHeadTp() {
+        List<?> authorsAndCollaborations = randomAuthorOrCollaborations();
         var head = new HeadTp();
+        head.getAuthorGroup().addAll(randomAuthorGroups(authorsAndCollaborations));
         head.setCitationTitle(randomCitationTitle());
         head.setAbstracts(randomAbstracts());
         head.setCitationInfo(randomCitationInfo());
-
         return head;
     }
 
+    private static Collection<? extends AuthorGroupTp> randomAuthorGroups(List<?> authorsAndCollaborations) {
+        List<AuthorGroupTp> authorGroupTps = new ArrayList<>();
+        int maxNumberOfAuthorGroups = 200;
+        var numbersOfAuthorGroups = randomInteger(maxNumberOfAuthorGroups) + 1;
```

Comment:
```
        return IntStream.range(0, randomInteger(maxNumberOfAuthorGroups))
            .boxed()
            .map(ignored -> randomAuthorGroup(authorsAndCollaborations))
            .collect(Collectors.toList());
```

Reply:
Nice, I hate for-loops with indexes particulary where you end up with "i", "j" etc., but I couldn't think of a way of doing it without it. This improves it alot. Thanks.

Side note: I think the array needs to have minimum size of 1. I can't imagine there being a publication without any authorgroups, especially since there's a collaboration type of child to authorgroup. So if you don't min I will go for do `range(0, randomInteger(maxNumberOfAuthorGroups) + 1)`